src/py/reactpy/reactpy/core/component.py

```diff
@@ -2,14 +2,15 @@
 
 import inspect
 from functools import wraps
-from typing import Any, Callable
+from typing import Any, Callable, TypeVar
 
 from reactpy.core.types import ComponentType, VdomDict
 
+T = TypeVar("T", bound=Callable[..., ComponentType | VdomDict | str | None])
 
 def component(
-    function: Callable[..., ComponentType | VdomDict | str | None]
-) -> Callable[..., Component]:
```

Comment:
I'm not sure the significance of going from `ComponentType | VdomDict | str | None` to `Component`. In my own code, I have a `Component` type which is more or less what the function is. I'll play with this on my environment and come back with changes or an A-OK to whether I am finding issues.